linalg/eigenvalue.hpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
```

Comment:
I wonder if this will be better in `solvers.?pp`... what do the reviewers think?

Reply:
In this form, it "lives" in its own world. It doesn't inherit from the `Solver` class and does not follow any pre-existent interface. On the other hand, it is in a way a "solver".

Maybe it can be put in the terms of the `Solver::Mult(...)` interface and return also the eigenvector, but I'm not a big fan of that idea.

If we look at already existing eigenslover, they seem to be in the matrix files and classes. Notably, the class `DenseMatrixEigensystem` is in `densemat.*`. So if we follow that paradigm, this needs to move to `operator.*`

Reply:
I don't think strongly about this either way...

Reply:
There are eigensolvers also in hypre and the densemat files

Reply:
Actually, yes, I'm for moving this to `operator.*`, but I also don't have strong opinion.

Reply:
I moved the `PowerMethod` class to `operator.*`